Sources/Diff/CollectionViewManager+Diff.swift

```diff
@@ -0,0 +1,261 @@
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+extension CollectionViewManager {
+
+    public typealias DiffCompletion = (Bool) -> Void
+
+    var diffSectionItems: [CollectionViewDiffSectionItem] {
+        return sectionItems.compactMap { sectionItem in
+            sectionItem as? CollectionViewDiffSectionItem
+        }
+    }
+
+    open func update(with sectionItems: [CollectionViewDiffSectionItem],
+                     diff: CollectionViewDiff,
```

Comment:
Is there a way to make it work without passing `CollectionViewDiff` each time? It feels more like a 'set & forget' type of variable. I don't think it'll be a common case to use different diff systems within a single app, let alone using different diffs within a single collection.

Reply:
It's a good point. We can use `diff` property in `CollectionViewManager`, but now all diff logic implemented in extension `CollectionViewManager+Diff` so I need to create associated object for that or we can move all diff logic to `CollectionViewManager`. What do you think?

Reply:
I don't know really. Given my 'love' for runtime manipulations, my first thought was to make a subclass `DiffableCollectionViewManager`, which will have it as a var, but yeah, it's less versatile that way.

Reply:
@artemnovichkov @stason314 @ReshetovE @Antowkos what do you think?

Reply:
There are a lot of runtime manipulations already, so one more manipulation shouldn't ruin everything

Reply:
agree with @Antowkos